src/actions/preferences.js

```diff
@@ -12,8 +13,10 @@ import {bindClientFunc} from './helpers';
 import {getProfilesByIds, getProfilesInChannel} from './users';
 import {getChannelAndMyMember, getMyChannelMember} from './channels';
 
-export function deletePreferences(userId, preferences) {
-    return async (dispatch, getState) => {
+import type {GetStateFunc, DispatchFunc, ActionFunc} from '../types/actions';
+
+export function deletePreferences(userId: string, preferences: Object): ActionFunc {
```

Comment:
In general the PR is ok, but you are using `Object` and we prefer to avoid that. If you want we can define the Preferences type. In this case, the preferences is an array of "Preference" type, which is
```
{|
  category: string
  name: string
  user_id: string
  value: string
|}
```